### ISSUE TITLE
How to deal with NAN values while optimizing in a very large space?

### ISSUE THREAD

**Issue by rogeravs1997** [keras-team/keras-tuner#477]
Hey everyone, I'm using keras tuner to build a model almost from scratch, optimizing n_layers, n_units, optimizer, parameters for optimizer and activation functions for every layer. This gives me a very large space to look up for optimization. Sometimes, the loss explodes and return nan values that stop the optimization. I was wondering how to deal with this error. I already discard "exponential" activation function, i choose to use "mae" as loss function, and also use "clipvalue" in every optimizer.

I have in mind two possible solutions: 
- Define a subclass for the BayesianOptimization class, with an inner check for nan values while updating the model.
- Clip the weights?

I hope someone can help me or give an intuition on what can i do.

**Comment by shun-lin** [comment]
have you tried Hyperband class? Also if you want to build model from scratch maybe AutoKeras can help, https://github.com/keras-team/autokeras.

**Comment by rogeravs1997** [comment]
> have you tried Hyperband class? Also if you want to build model from scratch maybe AutoKeras can help, https://github.com/keras-team/autokeras.

I'll check autokeras, I've never heard about it. Thank you so much :D

**Comment by rogeravs1997** [comment]
I solved by adding a TerminateonNan callback.

**Comment by haifeng-jin** [comment]
@rogeravs1997 Thank you for the solution!
I believe many people are facing this issue.
This is a neat solution before we fixing the bug.

**Comment by rogeravs1997** [comment]
Thank YOU for developing this great tool!
